sync-core/src/main/java/com/cloudant/sync/query/IndexUpdater.java

```diff
@@ -318,6 +318,12 @@ private DBParameter populateDBParameter(List<String> fieldNames,
             Object value = ValueExtractor.extractValueForFieldName(fieldName, rev.getBody());
 
             if (value != null) {
+                // Do not include an empty array field as you would not include a
+                // field with no value.
+                if (value instanceof List && ((List) value).size() == 0) {
+                    continue;
+                }
```

Comment:
If we ever include logic after the `if` statement, it's easy to miss that this `continue` statement would cause it to be ignored and introduce a bug. I'd reverse the condition and put the logic `includeFieldNames.add(fieldName); args.add(value);` inside the nested if.